clang/lib/DPCT/MigrateCmakeScript.cpp

```diff
@@ -737,18 +737,35 @@ void doCmakeScriptMigration(const clang::tooling::UnifiedPath &InRoot,
 }
 
 void registerCmakeMigrationRule(MetaRuleObject &R) {
-  auto PR =
-      MetaRuleObject::PatternRewriter(R.In, R.Out, R.Subrules, R.MatchMode,
-                                      R.RuleId, R.CmakeSyntax, R.Priority);
-
-  auto Iter = CmakeBuildInRules.find(PR.CmakeSyntax);
-  if (Iter != CmakeBuildInRules.end()) {
-    if (PR.Priority == RulePriority::Takeover) {
-      assert(Iter->second.Priority < PR.Priority &&
-             "Two same cmake syntaxes have \'Takeover\' priority.\n ");
+
+#ifdef __DEBUG_CMAKE_SCRIPR_MIGRATION
+  std::cout << "\n#####Migration Rule start ######" << std::endl;
+  std::cout << "\tRuleId:" << R.RuleId << std::endl;
+  std::cout << "\tCmakeSyntax:" << R.CmakeSyntax << std::endl;
+  std::cout << "\tIn:" << R.In << std::endl;
+  std::cout << "\tOut:" << R.Out << std::endl;
+  std::cout << "#####Migration Rule end ######" << std::endl;
+#endif
+  static std::unordered_set<std::string> DuplicateFilter;
+  if (DuplicateFilter.find(R.CmakeSyntax) == end(DuplicateFilter)) {
+    DuplicateFilter.insert(R.CmakeSyntax);
+
+    auto PR =
+        MetaRuleObject::PatternRewriter(R.In, R.Out, R.Subrules, R.MatchMode,
+                                        R.RuleId, R.CmakeSyntax, R.Priority);
+
+    auto Iter = CmakeBuildInRules.find(PR.CmakeSyntax);
+    if (Iter != CmakeBuildInRules.end()) {
+      if (PR.Priority == RulePriority::Takeover) {
+        assert(Iter->second.Priority < PR.Priority &&
```

Comment:
How about printing the rule ID of the duplicated rules?

Reply:
okay